wezterm-gui/src/termwindow/render/screen_line.rs

```diff
@@ -350,7 +351,12 @@ impl crate::TermWindow {
                 + (phys(params.cursor.x, num_cols, direction) as f32 * cell_width);
 
             if cursor_shape.is_some() {
-                let mut quad = layers.allocate(0).context("layers.allocate(0)")?;
+                let mut quad = match cursor_shape.unwrap() {
```

Comment:
in general, the bare `.unwrap()` is a bad code smell in Rust, especially right after a check like `.is_some()`.

A better way to phrase this block is:

```rust
if let Some(shape) = cursor_shape {
    let cursor_layer = match shape {
         // allow bar cursors to draw over the top of text
         CursorShape::BlinkingBar | CursorShape::SteadyBar => 2,
         // but all other cursors sit below it for legibility
         _ => 0,
    };
    let mut quad = layers.allocate(cursor_layer).with_context(||format!("layers.allocate({cursor_layer})"))?;
```

Reply:
Yes, but at the bottom (line ~412) we also pass `Option<CursorShape>` to `.cursor_spite()`. I'll do `Some(shape)`.